op-signer.md

```diff
@@ -0,0 +1,280 @@
+
+While in `infra/op-signer`, set the domain name or IP address of the server as the value of `SIGNER_SERVER_HOST` environment variable, and generate TLS:
+
+```bash
+export SIGNER_SERVER_HOST=65.108.236.27
+./tls-server.sh
+```
+
+You will receive several CA and TLS-related files in `tls-server` folder:
+
+```bash
+tls-server
+├── ca.crt
+├── ca.key
+├── ca.srl
+├── tls.crt
+├── tls.csr
+└── tls.key
+```
+> ⚠️ **Note:** 
+> If the CA key is compromised, an attacker can issue any certificate using it, which will be recognized as legitimate by the system.
+
+### Configure Google API Credentials
+
+Add the following line to `op-signer/.envrc` in the `infra` repo:
+
+```bash
+export GOOGLE_APPLICATION_CREDENTIALS="<PATH_TO_SERVICE_ACCOUNT_JSON_FILE>"
+```
+
+Replace `<PATH_TO_SERVICE_ACCOUNT_JSON_FILE>` with the path of the service account JSON file downloaded in [this step](#get-service-account-key-file).
+
+You can also change other configurations like port and log level in the `.envrc` file.
+
+### Configure Auth
+
+Modify the config.yaml file to connect op-signer with your cloud KMS:
+
+- **name**: DNS name of the client connecting to op-signer; must match the DNS name in the TLS certificate (`SIGNER_CLIENT_DNS`) used in [this step](#generate-client-tls).
+- **key**: key resource name from Cloud KMS obtained from [this step](#get-resource-name-of-the-key).
+
+For example:
+
+```yml
+auth:
+- name: op-challenger.beta.swc.quarkchain.io
+  key: projects/signing-test-450710/locations/global/keyRings/op-signer/cryptoKeys/op-challenger/cryptoKeyVersions/1
+```
+
+You can add multiple entries for different clients connecting with op-signer.
+
+### Retrieve Addresses
+
+For each key, there is a corresponding address that needs to be used as `from` when drafting Ethereum transactions. Execute this command to retrieve addresses from keys:
+
+```bash
+./bin/op-signer address
+```
+
+Output example:
+
+```
+0: projects/signing-test-450710/locations/global/keyRings/op-signer/cryptoKeys/op-challenger-1/cryptoKeyVersions/1 => 0x74D3b2A1c7cD4Aea7AF3Ce8C08Cf5132ECBA64ED
+```
+
+### Start op-signer Service
+
+In the `op-signer` folder, execute this command:
+
+```bash
+./bin/op-signer \
+--signer.tls.cert=./tls-server/tls.crt \
+--signer.tls.ca=./tls-server/ca.crt \
+--signer.tls.key=./tls-server/tls.key 
+```
+
+
+## Configure Signer Client
+
+### Generate Client TLS
+
+Pick a DNS for your client connecting to op-signer as value for `SIGNER_CLIENT_DNS`. This DNS should match one of those specified in your server's auth configuration (e.g., `op-challenger.beta.swc.quarkchain.io`). 
+
+Log on to the server where your op-signer service deployed in [this step](#build-the-source), and go to `op-signer` directory. Now set environment variable `SIGNER_CLIENT_DNS`, and generate TLS:
+
+```bash 
+export SIGNER_CLIENT_DNS=op-challenger.beta.swc.quarkchain.io 
+./tls-client.sh
+```
+
+You should now have these files in your `tls` folder:
+
+```bash 
+tls 
+├── tls.crt 
```

Comment:
If the hacker have the tls.key, can he request any signing data?

Reply:
If the tls connection is compromised, we may need the op-signer server side to control what is allowed to sign.